logging/loggingtest/logger.go

```diff
@@ -209,3 +211,27 @@ func (tl *Logger) Info(a ...interface{})             { tl.log(a...) }
 func (tl *Logger) Infof(f string, a ...interface{})  { tl.logf(f, a...) }
 func (tl *Logger) Debug(a ...interface{})            { tl.log(a...) }
 func (tl *Logger) Debugf(f string, a ...interface{}) { tl.logf(f, a...) }
+
```

Comment:
Not sure about this solution yet. Might be that I would change it to pass `Logger` explicitly to the package.

Reply:
the purpose of it is not clear to me. Where would it be used? What does the name 'Fire' mean?

Reply:
This is an implementation of so-called `Hook` for logrus logging lib (https://github.com/sirupsen/logrus#hooks). It is utilized in test setup: https://github.com/zalando/skipper/pull/1848/files#diff-e856a3236a5ba5505ac2fea89a1b7a62b9db3282be1cb8a00eed0ea9de177d2bR26 in order to "forward" all the logs from the logrus to to the test logger to be able to assert on it. I tested it as an alternative to passing logger directly to the `routesrv.RouteServer` for test purposes (as it is implemented in the `routing` library for example). It works fine. I exercised this option to check how it behaves, because:
- i would like to avoid passing a specific logger type (`logrus.Logger`) as an explicit dependency to our new library,
- why not already existing Skipper logger interface as type? (https://github.com/zalando/skipper/blob/master/logging/logger.go#L9); because it does not account for structure logging and I would like not to resign from structure logging, because of that,
- so alternatively new logging interface would have needed to be developed (might be just private to this lib) or the current one extended; which IMHO is a bigger decision to discuss.

Reply:
perfect, thanks for the explanation